### PR TITLE
chore: Metrics module added

### PR DESCRIPTION
**Pull request by hendrikebbers**
Fixes [#7654](https://github.com/hashgraph/hedera-services/issues/7654)

This PR creates a new module for the basic metrics api.

### CODE REVIEW

**Review by netopyr — DISMISSED** [review]

**Review by timo0 — DISMISSED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

**Review by imalygin — DISMISSED** [review]

**Review by leninmehedy — APPROVED** [review]

**Review by lpetrovic05 — APPROVED** [review]

**Review by timo0 — DISMISSED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by imalygin — DISMISSED** [review]

**Review by lpetrovic05 — APPROVED** [review]

**Review by timo0 — DISMISSED** [review]

**Review by timo0 — DISMISSED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

**Review by netopyr — DISMISSED** [review]
As per the latest announcement, we must not use the breaking change syntax in the title.

**Review by lpetrovic05 — APPROVED** [review]

**Review by jeromy-cannon — DISMISSED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Review by imalygin — APPROVED** [review]

**Review by Neeharika-Sompalli — DISMISSED** [review]
LGTM !

**Review by imalygin — APPROVED** [review]

**Review by cody-littley — DISMISSED** [review]

**Review by jeromy-cannon — APPROVED** [review]

**Review by rbair23 — DISMISSED** [review]

**Review by lpetrovic05 — APPROVED** [review]

File: platform-sdk/swirlds-metrics-api/src/main/java/com/swirlds/metrics/api/MetricTypes.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+package com.swirlds.metrics.api;
+
+public final class MetricTypes {
+
+    public static final String COUNTER = "COUNTER";
+
+    public static final String GAUGE = "GAUGE";
+
+    public static final String ACCUMULATOR = "ACCUMULATOR";
+}
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/main/java/com/swirlds/metrics/api/MetricTypes.java:26]: Might be worth adding "SPEEDOMETER" and "RUNNING_AVERAGE" here.

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/CounterTest.java [hunk 2]
```diff
@@ -77,7 +76,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(COUNTER);
+        assertThat(sut.getMetricType()).isEqualTo("COUNTER");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/CounterTest.java:79]: Perhaps compare against `MetricTypes.COUNTER` for consistency and reliability?

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/DoubleAccumulatorTest.java [hunk 3]
```diff
@@ -79,7 +78,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(ACCUMULATOR);
+        assertThat(sut.getMetricType()).isEqualTo("ACCUMULATOR");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/DoubleAccumulatorTest.java:81]: Might consider using `MetricTypes.ACCUMULATOR` here...

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/DoubleGaugeTest.java [hunk 4]
```diff
@@ -74,7 +73,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(GAUGE);
+        assertThat(sut.getMetricType()).isEqualTo("GAUGE");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/DoubleGaugeTest.java:76]: Perhaps use `MetricTypes.GAUGE` here?

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/IntegerAccumulatorTest.java [hunk 5]
```diff
@@ -79,7 +78,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(ACCUMULATOR);
+        assertThat(sut.getMetricType()).isEqualTo("ACCUMULATOR");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/IntegerAccumulatorTest.java:81]: Might consider using `MetricTypes.ACCUMULATOR` here...

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/IntegerGaugeTest.java [hunk 6]
```diff
@@ -74,7 +73,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(GAUGE);
+        assertThat(sut.getMetricType()).isEqualTo("GAUGE");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/IntegerGaugeTest.java:76]: Might consider using `MetricTypes.GAUGE` here...

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/LongAccumulatorTest.java [hunk 7]
```diff
@@ -79,7 +78,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(ACCUMULATOR);
+        assertThat(sut.getMetricType()).isEqualTo("ACCUMULATOR");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/LongAccumulatorTest.java:81]: Might consider using `MetricTypes.ACCUMULATOR` here...

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/LongGaugeTest.java [hunk 8]
```diff
@@ -74,7 +73,7 @@ public void reset() {}
 
     @Test
     void getMetricType() {
-        assertThat(sut.getMetricType()).isEqualTo(GAUGE);
+        assertThat(sut.getMetricType()).isEqualTo("GAUGE");
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/LongGaugeTest.java:76]: Might consider using `MetricTypes.GAUGE` here...

**Review by jsync-swirlds — APPROVED** [review]
Could make more use of the constants in `MetricsTypes`, but otherwise looks good.

**Review by jeromy-cannon — DISMISSED** [review]

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by lpetrovic05 — DISMISSED** [review]

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/DurationGauge.java [hunk 9]
```diff
@@ -35,8 +38,8 @@ public interface DurationGauge extends Metric {
      * {@inheritDoc}
      */
     @Override
-    default MetricType getMetricType() {
-        return MetricType.GAUGE;
+    default String getMetricType() {
```
Comment (netopyr) [inline platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/DurationGauge.java:41]: Returning a `String` instead of an enum seems like a step back. Why is this necessary?

Comment (hendrikebbers) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/metrics/DurationGauge.java:41]: because the basic metrics and the platform specific metrics are splitted now. Based one that an enum is not possible anymore

**Review by cody-littley — DISMISSED** [review]
Signing off on changes to hashgraph team owned files (it's mostly just import statement changes).

File: platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/LongGaugeTest.java (hunk 8 above)
Comment (hendrikebbers) [reply platform-sdk/swirlds-metrics-api/src/test/java/com/swirlds/metrics/api/LongGaugeTest.java:76]: MetricTypes reintroduced based on discussion.

**Comment by cody-littley** [comment]
@hendrikebbers I changed this to milestone 0.47, since I assume you don't intend to back port this change to mainnet 😉  (it was originally 0.45).

**Review by mxtartaglia-sl — APPROVED** [review]

**Review by jeromy-cannon — APPROVED** [review]

File: platform-sdk/swirlds-metrics-api/src/main/java/com/swirlds/metrics/api/MetricType.java [hunk 10]
```diff
@@ -14,11 +14,12 @@
  * limitations under the License.
  */
 
-package com.swirlds.common.metrics;
+package com.swirlds.metrics.api;
 
 /**
  * The types of {@code Metric}
  */
```
Comment (jsync-swirlds) [inline platform-sdk/swirlds-metrics-api/src/main/java/com/swirlds/metrics/api/MetricType.java:21]: Perhaps comment here regarding what should replace this, now deprecated, enum?
```
suggestion
 * @deprecated Please use <proper replacement for MetricType> instead.
 */
```

**Review by jsync-swirlds — APPROVED** [review]
Minor documentation improvement, otherwise looks even better.

**Review by netopyr — APPROVED** [review]
